.github/workflows/docker-build-push.yml

```diff
@@ -0,0 +1,40 @@
+name: docker-build-push
+on: [push, pull_request]
+
+jobs:
+  push_to_registry:
+    name: Build Docker image and push to Docker Hub
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Docker Metadata
+        id: docker_metadata
+        uses: docker/metadata-action@v3.8.0
+        with:
+          images: catalystcoop/pudl-etl
+          flavor: |
+            latest=auto
+          tags: |
+            type=ref,event=branch
+            type=ref,event=tag
+
+      - name: Set up Docker Buildx
```

Comment:
So this action makes buildx available for the "Build image and push to Docker Hub" step?

Reply:
It does that, but I think does other magical things too, or maybe buildx does those things on its own -- like creating a Docker build context that isn't the surrounding directory, and instead is derived directly from the git repository (which we then have to override below, to get access to the .git/ folder for `setuptools_scm`)